Services/DeviceModels.cs

```diff
@@ -153,5 +181,36 @@ private bool CheckStockDeviceModelExistence(string id)
             return this.stockDeviceModels.GetList()
                 .Any(model => id.Equals(model.Id, StringComparison.InvariantCultureIgnoreCase));
         }
+
+        /// <summary>
+        /// Transfrom property names to "Device.Reported.Properties"
+        /// </summary>
+        private void PreparePropertyNames(HashSet<string> set, object obj, string prefix)
```

Comment:
is all this code covered by unit tests? I couldn't see any specific one

Reply:
Done.